types/react-instantsearch-core/index.d.ts

```diff
@@ -707,11 +707,30 @@ interface HighlightResultPrimitive {
   fullyHighlighted?: boolean;
 }
 
+export type InsightsClient = (method: InsightsClientMethod, payload: InsightsClientPayload) => void;
+
+export type InsightsClientMethod = 'clickedObjectIDsAfterSearch' | 'convertedObjectIDsAfterSearch';
+
+export interface InsightsClientPayload {
+    index: string;
+    queryID: string;
+    eventName: string;
+    objectIDs: string[];
+    positions?: number[];
+}
+
+export interface ConnectHitInsightsProvided {
+    hit: Hit;
+    insights?: InsightsClient;
```

Comment:
Great work!  I think this parameter should be required, because that indicates that the component will always receive this prop.  Otherwise in strict mode, you'd have to do a test inside your component like `if (insights) { insights.doSomething() }`

Reply:
Ah yup I see what you mean. I'm not sure how to add that though since the `Hits` component can be used with or without insights ([re the docs on `Hits` which have an optional `hitComponent` prop](https://www.algolia.com/doc/api-reference/widgets/hits/react/)). If `insights` is required it would force everyone to pass in an insights client?

Reply:
You're correct, we don't want everyone to have to provide an insights client!  Fortunately, if I understand it correctly, it's the connector that is providing the insights client.  So the component can choose to either do something with that insights client or ignore it.  Typescript allows this!

The only thing Typescript wants is that one of the properties in your React components overlaps with one of the properties provided by the connector.

Reply:
I've made `insights` a required parameter now and used `Omit<>` in the function return type. Do you think this is the right approach?

Reply:
This looks great!  It definitely works well.  It's approved and can be merged :)

If you wanted to make one more improvement, you might consider using the `ConnectedComponentClass` type helper just as the other connectors do.  You can see an example with `connectHits`:

```ts
export function connectHits<THit = BasicDoc>(
  // tslint:disable-next-line:no-unnecessary-generics
  stateless: React.FunctionComponent<HitsProvided<THit>>
): React.ComponentClass;
export function connectHits<TProps extends HitsProvided<THit>, THit>(
  ctor: React.ComponentType<TProps>
): ConnectedComponentClass<TProps, HitsProvided<THit>>;

```

So instead of returning `React.ComponentType<Omit<ConnectHitInsightsProvided, { insights: WrappedInsightsClient }>>`, you would return `ConnectedComponentType<TProps, ConnectHitInsightsProvided>` which does the same thing.

Doing it this way would make the entire function generic over the props of the `hitComponent`.  Now I do understand that once you introduce generics, it makes things harder!  So if you play with it and it's not working, don't worry about it!  These types are certainly useful enough as they are.

Reply:
Thank you @gburgett ! I'll have a look at them but will probably take a while for me to figure out so I'll merge this in in the mean time